### ISSUE TITLE
Add support for testing AWS policy documents

### ISSUE THREAD

**Issue by Briansbum** [gruntwork-io/terratest#120]
Similar to how you can check an S3 bucket exists and confirm that it has the content you expect I'd like to be able to check that an IAM policy or S3 bucket policy exists.

I'd also like to check that it has the document contents I expect based on terraform outputs.

Testing this quickly I was able to use `json.Compact` on terraform's output and `assert.JSONeq` from [testify](https://godoc.org/github.com/stretchr/testify/assert#JSONEq) on the document return from these to test equality.

I don't know how to add these to the automated tests without adding complexity to them but if this isn't a worry I'd be happy to.

**Comment by brikis98** [comment]
This seems like a reasonable addition. That said, one of the core ideas with Terratest is not merely to assert that something was created in your environment, but that it actually behaves as it should (see [How Terratest compares to other testing tools](https://github.com/gruntwork-io/terratest#how-terratest-compares-to-other-testing-tools)). For example, checking that an IAM policy has the contents you expect is useful; but far more useful is performing various AWS operations and actually checking the IAM policy enforces the rules you want it to. If you had, say, an IAM policy that required MFA access for all actions, the "Terratest" way of testing that policy would be to try to make AWS API calls *without* an MFA token and ensuring those calls fail.

**Comment by Briansbum** [comment]
Considering how huge of a task that is perhaps I'll raise a PR just for the validation mentioned above but keep this issue open or open some more to address some jumping off points for testing the policies properly like you say.

I think a good place to start would be having tests for User/Role attachments that perform operations on s3. It'll be able to leverage the existing authentication methods in `aws/auth.go` and s3 doesn't have so many weird edge cases when it comes to api failures.

**Comment by brikis98** [comment]
I don't think we need to solve every possible validation right away. We build out Terratest by solving concrete use cases and add things to the library if they appear several times throughout our tests. If you find yourself verifying some behavior over and over, let us know, and it may be a good candidate for adding here.

**Comment by james03160927** [comment]
Closing this issue due to 6 years of inactivity. Feel free to reopen if it still needs to be addressed.
